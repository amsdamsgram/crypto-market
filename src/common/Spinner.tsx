import React from "react";
import { ActivityIndicator } from "react-native";
import styled from "styled-components/native";

import Theme from "../Theme";

enum Size {
  "small",
  "large"
}

interface IProps {
  size: Size;
}

const Container = styled.View`
  flex: 1;
  justify-content: center;
  padding: ${Theme.globalStyles.padding}px;
`;

const Spinner = (props: IProps) => {
  return (
    <Container>
      <ActivityIndicator size={props.size} />
    </Container>
  );
};

Spinner.defaultProps = {
  size: "large"
};

export default Spinner;
