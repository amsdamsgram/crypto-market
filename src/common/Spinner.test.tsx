import React from "react";
import renderer from "react-test-renderer";

import Spinner from "./Spinner";

describe("Spinner", () => {
  test("should match snapshot", () => {
    const tree = renderer.create(<Spinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
