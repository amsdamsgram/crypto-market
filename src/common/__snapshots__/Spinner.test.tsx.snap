// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
